### ISSUE TITLE
KAMA Instead Of EWMAPrice

### ISSUE THREAD

**Issue by asfbhero** [ctubio/Krypto-trading-bot#981]
https://sourceforge.net/projects/ta-lib/

Just throwing this out there for anyone who wants to work on this.  There is open source code in there that will calculate KAMA based on an input.

I tried adding this to the source, but this is honestly over my head.  Swapping out the calc function for this might be easy for someone who actually knows what they are doing.

I see this as valuable because as of now, if you get caught on the wrong side of a trend, you can stop yourself from taking any additional trades in the wrong direction, but I haven't found a way to get out of the bad trades quickly without sacrificing profitably when the market is ranging.

My current solution has been to trade pairs that are pretty much always ranging, which works, but I'm not sure how well this strategy will fare when you scale up the size.

With an adaptive moving average instead of an exponential one, it will stay right in the middle of the range when the market is flat, allowing you to do market making around the center of the range just as you would with a longer period EWMAPrice.  However, when the market starts trending you can have EWMATrend stop selling into an uptrend, then since the period of the moving average is effectively shorter during a fast moving market, you can get our of the losing shorts on the markets first dip to the KAMA.  Then just keep accumulating more buy's on the small dips, while EWMATrend is still preventing new sell trades.

**Comment by Camille92** [comment]
The maths of KAMA :

https://school.stockcharts.com/doku.php?id=technical_indicators:kaufman_s_adaptive_moving_average

So you want to replace (or offer the alternative) of ewma protection to KAMA ?

**Comment by asfbhero** [comment]
Yeah, replace/give an option to use KAMA instead of ewmaPrice protection.

**Comment by halessi** [comment]
I think this would be a fantastic addition. Not sure how similar it was to #948.

**Comment by asfbhero** [comment]
@halessi  It's close, but slightly different in that those ideas were geared more towards trend following using a custom indicator that used swings as a proxy for volatility which then lengthened or shortened the MA period.  Using KAMA I think is better as it is tested math, not something I cobbled together off a forum from 10 years ago.  

Also I think using KAMA instead of EWMAPrice is more suited to market making, which is more suited to what this bot is designed to do.

I believe the stuff I posted there still has value, but it would probably need lots of tweaking. Having both as an option would open up a lot of new possibilities, but if I had to choose the one that would increace profit the fastest, right out of the box, it would be KAMA instead of EWMAPrice.

**Comment by ctubio** [comment]
:koala: coOl''' many thanks for the suggestion''''

**Comment by nonkreon** [comment]
can somebody compare KAMA and MAMA, why KAMA specifically
